app/src/main/java/de/grobox/transportr/networks/PickTransportNetworkActivity.java

```diff
@@ -79,25 +80,55 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
 		}
 		setResult(RESULT_CANCELED);
 
-
 		adapter = new FastItemAdapter<>();
 		adapter.withSelectable(true);
-		adapter.getItemAdapter().withComparator(new RegionItem.RegionComparator(this));
 		adapter.withSelectionListener(this);
 		expandableExtension = new ExpandableExtension<>();
+		expandableExtension.withOnlyOneExpandedItem(true);
 		adapter.addExtension(expandableExtension);
 		list = findViewById(R.id.list);
 		list.setLayoutManager(new LinearLayoutManager(this));
 		list.setAdapter(adapter);
 
-		Map<Region, List<TransportNetwork>> networksByRegion = getTransportNetworksByRegion();
-		for (Region region : networksByRegion.keySet()) {
-			RegionItem regionItem = new RegionItem(region);
-			List<TransportNetwork> networks = networksByRegion.get(region);
-			List<TransportNetworkItem> networkItems = new ArrayList<>(networks.size());
-			for (TransportNetwork n : networks) networkItems.add(new TransportNetworkItem(n));
-			regionItem.withSubItems(networkItems);
-			adapter.add(regionItem);
+		Map<Region, List<Region>> regions = getRegionsByRegion();
+		List<ContinentItem> continentItems = new ArrayList<>(Continent.values().length);
+		for (Continent continent : Continent.values()) {
+			if (!regions.containsKey(continent)) {
+				continue;
+			}
+			ContinentItem continentItem = new ContinentItem(continent);
+			List<Region> subRegions = regions.get(continent);
+			List<RegionItem> subRegionItems = new ArrayList<>(subRegions.size());
+			for (Region subRegion : subRegions) {
+				if (subRegion instanceof Country) {
+					Country country = (Country)subRegion;
+					if (!regions.containsKey(country)) {
+						continue;
+					}
+					CountryItem countryItem = new CountryItem(country);
+					List<Region> networks = regions.get(country);
+					List<TransportNetworkItem> networkItems = new ArrayList<>(networks.size());
+					for (Region network : networks) {
```

Comment:
This triple-nested loop looks a bit crazy. I wonder if we can't simply things a bit so it is less complicated and easier to understand.

Reply:
Maybe when the networks are defined, we could already define the hierarchy (best in Kotlin), so we save ourselves these iterations?

Reply:
> Maybe when the networks are defined, we could already define the hierarchy (best in Kotlin), so we save ourselves these iterations?

This could simplify it a lot, indeed. Where would be the right place for it then?

Reply:
We would probably do this right in the `TransportNetworks` interface. But maybe as a second step after this PR is done.